### PR TITLE
Remove getCourseDetails method from CourseLogic #5402

### LINKED ISSUE #5402: getCourseDetails method in CoursesLogic could be removed
`getCourseDetails` just calls to `getCourseSummary` and doesn't add any extra functionality. All calls could be replaced by `getCourseSummary`

### PR DESCRIPTION
**Pull request by LimeTheCoder**
Fixes #5402

### CODE REVIEW

File: src/test/java/teammates/test/cases/logic/CoursesLogicTest.java [hunk 1]
```diff
@@ -374,7 +375,8 @@ public void testGetCourseDetails() throws Exception {
         ______TS("null parameter");
 
         try {
-            coursesLogic.getCourseDetails(null);
+            String nullableCourseId = null;
```
Comment (Neurrone) [inline src/test/java/teammates/test/cases/logic/CoursesLogicTest.java]: Suggest renaming NullableCourseId to nullCourseId. Nullable implies that null is a valid state for the course id.

**Comment by LimeTheCoder** [comment]
Changes made.

File: src/test/java/teammates/test/cases/logic/StudentsLogicTest.java [hunk 2]
```diff
@@ -651,7 +651,8 @@ public void testEnrollStudents() throws Exception {
         verifyEnrollmentResultForStudent(saf.makeStudent(line4, courseIdForEnrollTest),
                                          enrollResults.studentList.get(4), StudentUpdateStatus.NEW);
             
-        CourseDetailsBundle cd = coursesLogic.getCourseDetails(courseIdForEnrollTest);
+        CourseDetailsBundle cd = coursesLogic
+                .getCourseSummary(courseIdForEnrollTest);
```
Comment (samsontmr) [inline src/test/java/teammates/test/cases/logic/StudentsLogicTest.java]: These two lines should be on one line

File: src/test/java/teammates/test/cases/logic/CoursesLogicTest.java [hunk 3]
```diff
@@ -328,7 +328,8 @@ public void testGetCourseDetails() throws Exception {
         ______TS("typical case");
 
         CourseAttributes course = dataBundle.courses.get("typicalCourse1");
-        CourseDetailsBundle courseDetails = coursesLogic.getCourseDetails(course.getId());
+        CourseDetailsBundle courseDetails = coursesLogic
+                .getCourseSummary(course.getId());
```
Comment (samsontmr) [inline src/test/java/teammates/test/cases/logic/CoursesLogicTest.java]: These two lines should be on one line

File: src/test/java/teammates/test/cases/logic/StudentsLogicTest.java (hunk 2 above)
Comment (LimeTheCoder) [inline src/test/java/teammates/test/cases/logic/StudentsLogicTest.java]: I tried to follow JCC, where max line size - 80, but now I see that you use code convention from google, where limit per line - 100 symbols.

**Pull request merged**